CHANGELOG.md

```diff
@@ -1,6 +1,129 @@
+    - Empty samples and/or features being automatically removed
+    - Features in the table that weren't in the feature rankings being
+      automatically removed
+    - Samples in the table that weren't in the sample metadata being
+      automatically removed (the opposite situation -- samples in the metadata
+      that weren't in the table -- already would trigger output messages,
+      though).
+    - More context regarding the details of feature filtering, if specified via
+      the `-x`/`--extreme-feature-count`/`--p-extreme-feature-count` option.
+  These output messages will show up normally when running Qurro standalone or
+  when using the `--verbose` option when running Qurro through QIIME 2.
+- Added a `--p-debug` option when running Qurro through QIIME 2. This will
+  cause debug messages to be output when the `--verbose` flag has also been
+  specified to QIIME 2. (Non-debug output messages -- for example, the print
+  messages described above -- will be output when `--verbose` has been
+  specified to QIIME 2, but regardless of whether or not `--p-debug` has been
+  specified.)
+  ([#95](https://github.com/biocore/qurro/issues/95))
+### Backward-incompatible changes
+- Renamed Qurro's QIIME 2 actions ([#98](https://github.com/biocore/qurro/issues/98)):
+    - **Renamed `qiime qurro supervised-rank-plot` to `qiime qurro differential-plot`.**
+    - **Renamed `qiime qurro unsupervised-rank-plot` to `qiime qurro loading-plot`.**
+- Replaced `-v`/`--verbose` in Qurro's standalone script with `--debug`, in
+  order to be consistent with Qurro's QIIME 2 plugin.
+### Bug fixes
+- A clear error is now raised if the feature rankings or feature metadata
+  include a column named `qurro_x`. (This column name is used internally by
+  Qurro to sort features in the rank plot.)
+  ([#183](https://github.com/biocore/qurro/issues/183))
+  Before, input datasets with this column name would have caused either the
+  loss of this column of data or confusing errors.
+- Previously, there was a small (either almost or entirely impossible) chance
+  that all of the main data within the rank or sample plot JSON could get
+  overwritten due to a collision in the dataset name. Although this should
+  basically never happen anyway, Qurro's code now checks for this scenario and
+  raises an error accordingly.
+  ([#190](https://github.com/biocore/qurro/issues/190))
+### Performance enhancements
+### Miscellaneous
+- Updated the rank plot's y-axis to say `Magnitude: [ranking name]` instead of
+  `Rank: [ranking name]`.
+  ([#194](https://github.com/biocore/qurro/issues/194))
+- Added a draft "Moving Pictures" tutorial.
+- DEICODE ordinations used in demos and test inputs have been rerun with
+  DEICODE version 0.2.3.
+  ([#188](https://github.com/biocore/qurro/issues/188))
+- Cleaned up Qurro's command-line interface options and help text.
+- A clear error is now raised if Qurro is trying to parse a GNPS feature
```

Comment:
Why does this cause an error?

Reply:
Qurro stores feature rankings and feature metadata in the same JS objects. There might be a better way around this, but this was the easiest thing to do that allowed us to use both types of values in the same context (e.g. for rank plot tooltips, or for searching through these fields).

This means that the feature ranking and feature metadata names can't conflict at all. This has already been enforced using `check_column_names()` ([1](https://github.com/biocore/qurro/blob/8b4a1b7e008fb3a433807597c2d9f491b0815c5e/qurro/generate.py#L128), [2](https://github.com/biocore/qurro/blob/8b4a1b7e008fb3a433807597c2d9f491b0815c5e/qurro/_df_utils.py#L385)), but IIRC I wasn't really enforcing this sorta stuff for the GNPS feature metadata until this update. Or at least if I was then it would result in a strange error message. It should be taken care of now.